lua/entities/gmod_wire_expression2/core/custom/cfc_vector2_NumberOperands.lua

```diff
@@ -0,0 +1,396 @@
+local delta  = wire_expression2_delta
+
+local math   = math
+local random = math.random
+local pi     = math.pi
+local inf    = math.huge
+
+local exp    = math.exp
+local frexp  = math.frexp
+local log    = math.log
+local log10  = math.log10
+local sqrt   = math.sqrt
+local abs    = math.abs
+
+local floor  = math.floor
+local ceil   = math.ceil
+local Round  = math.Round
+
+local sin    = math.sin
+local cos    = math.cos
+local tan    = math.tan
+
+local acos   = math.acos
+local asin   = math.asin
+local atan   = math.atan
+local atan2  = math.atan2
+
+local sinh   = math.sinh
+local cosh   = math.cosh
+local tanh   = math.tanh
+
+__e2setcost(1)
+
+registerOperator("exp", "xv2xv2", "xv2", function(self, args)
+	local op1, op2 = args[2], args[3]
+	return {
+		op1[1](self, op1) ^ op2[1](self, op2),
+		op1[2](self, op1) ^ op2[2](self, op2),
+	}
+end)
+
+
+__e2setcost(2) -- approximation
+
+--- Returns true (1) if given value is a finite number; otherwise false (0).
+e2function vector2 isfinite(vector2)
+	return {
+		(vector2[1] > -inf and vector2[1] < inf) and 1 or 0,
+		(vector2[2] > -inf and vector2[2] < inf) and 1 or 0
+	}
+end
+
+--- Returns 1 if given value is a positive infinity or -1 if given value is a negative infinity; otherwise 0.
+e2function vector2 isinf(vector2)
+	local function numIsInf(number)
+		if number == inf then return 1 elseif number == -inf then return -1 end
+		return 0
+	end
+	return {
+		numIsInf(vector2[1]),
+		numIsInf(vector2[2])
+	}
+end
+
+--- Returns true (1) if given value is not a number (NaN); otherwise false (0).
+e2function vector2 isnan(vector2)
+	local function isNaNNum(number)
+		return (value ~= value) and 1 or 0
+	end
+	return {
+		isNaNNum(vector2[1]),
+		isNaNNum(vector2[2])
+	}
+end
+
+--[[************************************************************************]]--
+
+__e2setcost(2) -- approximation
+
+e2function vector2 abs(vector2)
+	return {
+		abs(vector2[1]),
+		abs(vector2[2])
+	}
+end
+
+--- rounds towards zero
+e2function vector2 int(vector2)
+	local function intNum(number)
+		if rv1 >= 0 then return floor(rv1) else return ceil(rv1) end
+	end
+	return {
+		intNum(vector2[1]),
+		intNum(vector2[2])
+	}
+end
+
+--- returns the fractional part. (frac(-1.5) == 0.5 & frac(3.2) == 0.2)
+e2function vector2 frac(vector)
+	local function fracnum(number)
+		if rv1 >= 0 then return rv1 % 1 else return rv1 % -1 end
+	end
+	return {
+		fracnum(vector[1]),
+		fracnum(vector[2])
+	}
+end
+
+registerFunction("wrap", "xv2xv2", "xv2", function(self, args)
+	local op1, op2, op3, op4 = args[2][1], args[2][2], args[3][1], args[3][2]
+	local rv1, rv2, rv3, rv4 = op1[1](self, op1), op2[1](self, op2), op3[1](self, op3), op4[1](self, op4)
+	return {
+		(rv1 + rv3) % (rv3 * 2) - rv3,
+		(rv2 + rv4) % (rv4 * 2) - rv4
+	}
+end)
+
+--[[************************************************************************]]--
+
+__e2setcost(2) -- approximation
+
+registerFunction("sqrt", "xv2", "xv2", function(self, args)
```

Comment:
would it be possible to use`the `e2function type name()` syntax here and for the other registerFunction calls

Reply:
probably, but it probably also works this way.

Reply:
just because it works doesn't mean it should necessarily be done that way

Reply:
If you really want it that way.

Reply:
Done

Reply:
Dope